### ISSUE TITLE
app.onError/app.notFound do not have the correct type for Context param

### ISSUE THREAD

**Issue by Skye-31** [honojs/hono#511]
```ts
export interface MyEnv {
	Bindings: {
		NAMESPACE: KVNamespace;
	}
}
const app = new Hono<MyEnv>();
app.onError((err, c) => {
  // c is typed as Context<string, Environment>, not Context<string, MyEnv>, resulting in typescript errors when trying to access my namespace
  // this also applies to app.notFound
  //to access my types, I have to do:
  const context = c as unknown as Context<"", Env>;
})

**Comment by yusukebe** [comment]
Hi @Skye-31 !

This issue may be fixed by #512 

The next version will include this patch, please wait!

**Comment by Skye-31** [comment]
This looks to fix it, thanks!
